### ISSUE TITLE
Some parts of the chm manual file are in a very small font

### ISSUE THREAD

**Issue by lkintact** [php/doc-en#3780]
I downloaded the [chm version of the manual](https://www.php.net/distributions/manual/php_enhanced_en.chm) and found that some parts of it are in a very small font, they are underlined in red in this picture (that's at "PHP Manual -> Function reference -> Image Processing and Generation -> ImageMagick -> Installing/Configuring -> Installation):
![chm_manual_small_font](https://github.com/user-attachments/assets/f8593169-14f1-4b12-a65b-e813e5cbfd14)

**Comment by Girgias** [comment]
Maybe @mumumu has an idea? As IIRC they know how to build the CHM version.

But this might be an issue with PhD our doc renderer.

**Comment by jimwins** [comment]
This is probably a missing style for the CHM version (which basically just a particular HTML format). Looks like it is for anything that ends up in a `<var>` tag, so filenames, environment variables, and other variables. I can blindly change the stylesheet for the CHM version (`chm/make_chm_style.css` in `doc-base`, I think) to see if it helps, but I don't have any Windows systems to test it on. (There's a style for 'PRE', I'd make it 'VAR, PRE' to see if that helps.)

Looks like @bjori is involved in the building of the CHM versions, they may have more insight.

**Comment by bjori** [comment]
Sounds like a good theory @jimwins . Give it a try and see if @lkintact can verify if it worked.
I don't think I ever had first-hand experiencing using the chm builds :)

**Comment by jimwins** [comment]
Looking at it more, it's actually probably one of the CSS files in the `htmlhelp/suppfiles` directory of `doc-base`. But I'm not even sure how/when this gets rebuilt because the CHM files were last updated about two months ago.

It should be possible to use GitHub Actions to build these. There's a [command-line HTMLHelp compiler](https://learn.microsoft.com/en-us/previous-versions/windows/desktop/htmlhelp/to-compile-a-help-project-from-the-dos-prompt) so then it's "just" a matter of building the proper input, running that, and making the output available somewhere. That's going to require someone with more active Windows knowledge than I have, though.

**Comment by cmb69** [comment]
I've downloaded php_manual_en.chm, and it looks fine in HTML Control 10.0:

![chm](https://github.com/user-attachments/assets/07474928-6498-465f-9116-2a62a49db044)

It looks the same if I build that file manually.

I can imagine the problem is that the Source Code Pro font is not installed on the machine. I think it's best to use standard fonts for the CHMs (or to embed the fonts if allowed and possible).

And frankly, I'm not sure whether CHM is future proof; I didn't even find a downloadable help compiler (but had it installed already, after looking more closely), and apparently MS considers the format obsolete. Maybe PDF is the way to go.

**Comment by jimwins** [comment]
Strange, I thought I saw the download for the tools just recently, but poking around more it seems like the only way to get them is the Wayback Machine or other sketchy links.

If we don't want to create a way for us to build the CHM manuals, it's probably time to retire them. PhD doesn't currently work for PDF, but that's not really a close analogue to CHM, anyway. Maybe ePub would be a good alternative? The current tooling seems to work for generating it.

I don't think there's a way we can do something like the CHM with user notes that we had that would be worth the effort in doing so.

**Comment by cmb69** [comment]
Yeah, maybe ePub is the better alternative. I'm more concerned about https://github.com/php/systems/issues/58 for now.

**Comment by mumumu** [comment]
I'm aware of this problem myself, it's caused by css not working properly anymore.

The reason chm hasn't been able to update for 2 months is because my local machine was down. Sorry, I can resume updating as soon as you're ok with the version without css running.

But as @cmb69 pointed out, chm is obviously legacy; the relevant binaries needed to build chm are already unavailable for download, and I only have what I have on hand. I'll continue to fix this issue and update chm if there's demand, but if you can migrate to PDF or epub, I'd be happy to do so.

**Comment by cmb69** [comment]
I had a look at this again, and apparently, there is almost no CSS involved (except for [style.css](https://github.com/php/phd/blob/673b2dab11883de1f93ce74683b628e26865cce2/phpdotnet/phd/Package/PHP/CHM.php#L250-L254)). Instead, the problem seems to be related to the `<var>` elements; on my machine they render fine, but that might depend on the reader (and the underlying rendering engine), and given that `<var>` is relatively new, it might just be not properly recognized. So if we want to stick with minimal styling, we should probably replace `<var>` with `<code>` or something.

Or we try to bring back proper stylesheets, which are supported by the HTML help compiler. There is of course no guarantee that these styles would be properly supported by all readers (although ePub likely has the same issue).

**Comment by cmb69** [comment]
I have manually replaced all `var(…)` in the stylesheets with their values, and removed the `:root` selector. Afterwards, HTML help workshop did no longer complain about the stylesheets, and here the result looks like:

![chm](https://github.com/user-attachments/assets/0a5d2832-3939-4d04-b655-aeb091b50d61)

It might be worthwhile to automate this step, so we can ship with full CSS again.
